src/shogun/machine/gp/GaussianARDFITCKernel.cpp

```diff
@@ -98,42 +100,45 @@ SGMatrix<float64_t> CGaussianARDFITCKernel::get_parameter_gradient(
 		int32_t idx_l=index;
 		//Note that CDotKernel requires lhs and rhs are CDotFeatures pointers
 		//This Kernel is a subclass of CDotKernel
-		SGVector<float64_t> left_vec=((CDotFeatures *)lhs)->get_computed_dot_feature_vector(idx_l);
+		SGVector<float64_t> left_vec=get_feature_vector(idx_l, lhs);
 		SGMatrix<float64_t> res(left_vec.vlen, num_rhs);
 
 		for (int32_t idx_r=0; idx_r<num_rhs; idx_r++)
 		{
-			SGVector<float64_t> right_vec=((CDotFeatures *)rhs)->get_computed_dot_feature_vector(idx_r);
+			SGVector<float64_t> right_vec=get_feature_vector(idx_r, rhs);
 			SGMatrix<float64_t> res_transpose(res.get_column_vector(idx_r),1,left_vec.vlen,false);
```

Comment:
I thought a bit about this and it turns out that in-place is not possible if we allow implicit conversion to/from SG/CGPU. I need to figure out a proper way to do this. It is better that you use eigen3. Else if you do
```c++
right_vec = linalg::add(left_vec, right_vec, ...);
```
this would technically work but you'll end up reallocating+freeing a chunk.